repos/spack_repo/builtin/packages/py_astropy/package.py

```diff
@@ -32,20 +33,29 @@ class PyAstropy(PythonPackage):
 
     variant("all", default=False, when="@3.2:", description="Enable all functionality")
 
-    depends_on("c", type="build")  # generated
+    depends_on("c", type="build")
 
     # Required dependencies
+    depends_on("python@3.11:", when="@7.0.1:", type=("build", "run"))
     depends_on("python@3.10:", when="@6.1.0:", type=("build", "run"))
     depends_on("python@3.8:", when="@5.1:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("py-cython@0.29.13:", type="build")
     depends_on("py-cython@0.29.30", when="@5.1:6.0", type="build")
-    depends_on("py-cython@3.0.0", when="@6.1.0:", type="build")
+    depends_on("py-cython@3.0.0", when="@6.1.0:6", type="build")
```

Comment:
in pyproject.toml, cython dependency is indicated as `"cython>=3.0.0,<3.1.0",` for version @6.1:6. it would be nice test that package actually builds with all cython@3.0.x versions.
https://github.com/astropy/astropy/blob/v6.0.0/pyproject.toml